Fixed and improved docs

rohitsinha54 opened the pull request:
- Made docs consistent with Code
- Replaced commands to use cdap-cli
- Updated and tested all commands to work.

Build : https://builds.cask.co/browse/CG-SGUT1-1

Commit:
Fixed and improved docs

abaranau commented on README.md:
```diff
@@ -80,14 +80,14 @@ public class PageRankApp extends AbstractApplication {
+    setName(PageRankApp.class.getSimpleName());
     addSpark(new PageRankSpark());
-    addService("PageRankService", new PageRankHandler());
+    addStream(new Stream(PAGE_RANK_BACKLINK_STREAM));
```
hm, seems like code became less readable - now you are missing the constants... Any reason for this change?

abaranau replied:
also: you don't have to prefix all constants with PAGE_RANK

rohitsinha54 replied:
I don't want to hard code the stream name here. 
I see the stream name in the unit test so I can reuse the constant there. That is the reason why I made it this way.

awholegunch replied:
you don't have to prefix all constants with PAGE_RANK: +1

abaranau commented on README.md:
```diff
@@ -118,7 +118,7 @@ public class PageRankSpark extends AbstractSpark {
   public SparkSpecification configure() {
     return SparkSpecification.Builder.with()
-      .setName("PageRankProgram")
+      .setName(PageRankProgram.class.getSimpleName())
```
this looks really hacky. Better string or constant called NAME

rohitsinha54 replied:
I don't understand why it looks hacky. 
In this case and almost all others we generally keep the name as the class name itself and its just one more thing in the code. In this way we have it same and if the class name changes for any reason it will get reflected everywhere. 
Don't you agree ?

abaranau commented on README.md:
```diff
@@ -216,34 +213,34 @@ You can then deploy the application to a standalone CDAP installation:
 for completion using:
 
-    curl -v 'http://localhost:10000/v2/apps/PageRankApp/spark/PageRankProgram/status'
+    cdap-cli.sh get spark status PageRankApp.PageRankProgram
```
nice!

abaranau commented on README.md:
```diff
@@ -216,34 +213,34 @@ You can then deploy the application to a standalone CDAP installation:
 Run the Spark Program:
 
-    curl -v -X POST 'http://localhost:10000/v2/apps/PageRankApp/spark/PageRankProgram/start'
+    cdap-cli.sh start spark PageRankApp.PageRankProgram
```
nice!

abaranau commented on README.md:
```diff
@@ -216,34 +213,34 @@ You can then deploy the application to a standalone CDAP installation:
 Query for the PageRank results:
 
-    curl -v -d 'http://example.com/page10' 'http://localhost:10000/v2/apps/PageRankApp/services/PageRankService/methods/pagerank'
+    curl -v 'http://localhost:10000/v2/apps/PageRankApp/services/PageRankService/methods/pagerank?url=http://example.com/page1'
```
do we have JIRA for calling service endpoints thru CLI?

awholegunch replied:
I think Alvin has been working on this; I'll see if it that's the case.

awholegunch replied:
There is now: https://issues.cask.co/browse/CDAP-855

abaranau commented on README.md:
```diff
@@ -216,34 +213,34 @@ You can then deploy the application to a standalone CDAP installation:
 Example output:
 
-    0.45521228811700043
+    0.2610116705534049
```
improved!

rohitsinha54 replied:
The pages are different :D 
page1 is the first one so it looks more simpler :)

abaranau commented on src/main/java/co/cask/cdap/guides/PageRankSpark.java:
```diff
@@ -35,7 +35,7 @@
   public SparkSpecification configure() {
     return SparkSpecification.Builder.with()
-      .setName(PageRankSpark.class.getSimpleName())
+      .setName(PageRankProgram.class.getSimpleName())
```
Does it make sense to call it PageRankSparkProgram - will help to read and will get immediate attention of those who are looking for how to implement SparkProgam and run in CDAP

rohitsinha54 replied:
Agree.
Now I have to update the diagram and more docs :cold_sweat:

abaranau commented:
Cool! See comments

awholegunch commented on src/test/java/co/cask/cdap/guides/PageRankAppTest.java:
```diff
@@ -59,7 +59,7 @@ public void test() throws Exception {
+    SparkManager sparkManager = appManager.startSpark(PageRankProgram.class.getSimpleName());
     sparkManager.waitForFinish(60, TimeUnit.SECONDS);
 
     ServiceManager serviceManager = appManager.startService(PageRankApp.PAGE_RANK_RANKS_SERVICE);
```
Can we improve these names? I think "PageRankApp.PAGE_RANK_RANKS_SERVICE" is a bit of a mouthful. What about "PageRankApp.RANKS_SERVICE"?

Commit:
Addressed comments

Commit:
More fixes

rohitsinha54 commented:
@abaranau : I have fixed according to the comments. 
Also renamed the Spark scala program to PageRankSparkProgram. Can you please just change the name of Spark in the docs  image. Since its a graffle file and I don't have license I cant open it. Back to our earlier problem :)

Build running here: https://builds.cask.co/browse/CG-SGUT1-4

@awholegunch : Please have a quick look too.

awholegunch commented:
LGTM, though the image must be updated...

abaranau commented on README.md:
```diff
@@ -118,15 +118,15 @@ public class PageRankSpark extends AbstractSpark {
+      .setName(PageRankSpark.class.getSimpleName())
       .setDescription("Spark program to compute PageRank")
-      .setMainClassName(PageRankProgram.class.getName())
+      .setMainClassName(PageRankSparkProgram.class.getName())
```
let's have a constant as we disscussed, this looks hacky

rohitsinha54 replied:
This cannot be a  constant was this needs to be the full qualified name of the class which is user's spark program.

rohitsinha54 replied:
Our discussion was for the .setName () I am making that a constant.

abaranau commented on src/main/java/co/cask/cdap/guides/PageRankSpark.java:
```diff
@@ -37,7 +37,7 @@ public SparkSpecification configure() {
       .setName(PageRankSpark.class.getSimpleName())
       .setDescription("Spark program to compute PageRank")
-      .setMainClassName(PageRankProgram.class.getName())
+      .setMainClassName(PageRankSparkProgram.class.getName())
```
constant, otherwise hacky

abaranau commented:
please fix last comment and MERGE

Commit:
Updated acc to comments

rohitsinha54 commented:
Fixed. Build running here: https://builds.cask.co/browse/CG-SGUT1-5

rohitsinha54 commented:
Build passed. Merging.
